add note on write access to socket_path (#67060)
ansible/ansible: docs/docsite/rst/network/user_guide/network_debug_troubleshooting.rst

.. _socket_path_issue:

Category "socket path issue"
============================

**Platforms:** Any

> The ``socket path does not exist or cannot be found``  and ``unable to connect to socket`` messages are new in Ansible 2.5. These messages indicate that the socket used to communicate with the remote network device is unavailable or does not exist.

Reviewer: ```suggestion
The ``Socket path does not exist or cannot be found``  and ``Unable to connect to socket`` messages are new in Ansible 2.5. These messages indicate that the socket used to communicate with the remote network device is unavailable or does not exist.
```

> The ``socket_path does not exist or cannot be found``  and ``unable to connect to socket`` messages are new in Ansible 2.5. These messages indicate that the socket used to communicate with the remote network device is unavailable or does not exist.

Reviewer: ```suggestion
 The ``socket path does not exist or cannot be found``  and ``unable to connect to socket`` messages are new in Ansible 2.5. These messages indicate that the socket used to communicate with the remote network device is unavailable or does not exist.
```

   fatal: [spine02]: FAILED! => {
       "changed": false,
       "failed": true,

>        "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_TSqk5J/ansible_modlib.zip/ansible/module_utils/connection.py\", line 115, in _exec_jsonrpc\nansible.module_utils.connection.ConnectionError: socket path XX does not exist or cannot be found\n",

Reviewer: ```suggestion
       "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_TSqk5J/ansible_modlib.zip/ansible/module_utils/connection.py\", line 115, in _exec_jsonrpc\nansible.module_utils.connection.ConnectionError: Socket path XX does not exist or cannot be found. See the socket path issue category in Network Debug and Troubleshooting Guide\n",
```

   fatal: [spine02]: FAILED! => {
       "changed": false,
       "failed": true,

>        "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_TSqk5J/ansible_modlib.zip/ansible/module_utils/connection.py\", line 123, in _exec_jsonrpc\nansible.module_utils.connection.ConnectionError: unable to connect to socket XX\n",

Reviewer: ```suggestion
       "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_TSqk5J/ansible_modlib.zip/ansible/module_utils/connection.py\", line 123, in _exec_jsonrpc\nansible.module_utils.connection.ConnectionError: Unable to connect to socket XX. See the socket path issue category in Network Debug and Troubleshooting Guide\n",
```

Suggestions to resolve:

> #. Verify that you have write access to the ``socket_path``.

Reviewer: I don't know that `socket_path` is what we should be calling out here. Although, now that I look at the context, that might be a lost cause.

The issue I see is that `socket_path` isn't a thing that the user sets or knows about. The only user-facing references to it that I can find are these error messages and they don't even mention what the value of socket_path is. I think we need to re-evaluate what those errors say before this instruction can be any help.

> #. Verify that you have write access to the ``socket_path``.

Reviewer: ```suggestion
#. Verify that you have write access to the path described in the error message
```
